nimare/workflows/cluster.py

```diff
@@ -1,8 +1,113 @@
-from ..meta.cbma.kernel import ALEKernel
-from ..due import due, Doi
+import click
+import numpy as np
+import pandas as pd
+import seaborn as sns
+import nibabel as nib
+import matplotlib.pyplot as plt
+from sklearn.cluster import KMeans, SpectralClustering, DBSCAN
+from sklearn.metrics import silhouette_score
 
+#from ..due import due, Doi
+#from ..meta.cbma import kernel
+#from ..dataset.extract import convert_sleuth_to_database
 
-@due.dcite(Doi('10.1073/pnas.0905267106'),
-           description='Introduces meta-analytic clustering analysis.')
-def meta_cluster_workflow(dataset, n_clusters=[5]):
-    pass
+from nimare.dataset.extract import convert_sleuth_to_database
+from nimare.meta.cbma.kernel import ALEKernel, MKDAKernel, KDAKernel, Peaks2MapsKernel
+from nimare.due import due, Doi
+
+
+@click.command(name='metacluster')
+@click.argument('database', required=True, type=click.Path(exists=True, readable=True))
+@click.option('--output_dir', required=True, type=click.Path(exists=True), help='Directory into which clustering results will be written.')
+@click.option('--output_prefix', default='metacluster', type=str, help='Common prefix for output clustering results.')
+@click.option('--kernel', default='ALEKernel', type=click.Choice(['ALEKernel', 'MKDAKernel', 'KDAKernel', 'Peaks2MapsKernel']), help='Kernel estimator, for coordinate-based metaclustering.')
+@click.option('--coord/--img', required=True, default=False, help='Is input data image- or coordinate-based?')
+@click.option('--algorithm', '-a', default='kmeans', type=click.Choice(['kmeans', 'dbscan', 'spectral']), help='Clustering algorithm to be used, from sklearn.cluster.')
+@click.option('--clust_range', nargs=2, type=int, help='Select a range for k over which clustering solutions will be evaluated (e.g., 2 10 will evaluate solutions with k = 2 clusters to k = 10 clusters).')
+@due.dcite(Doi('10.1016/j.neuroimage.2015.06.044'),
+           description='Introduces meta-analytic clustering analysis; hierarchically clusering face paradigms.')
+@due.dcite(Doi('10.1162/netn_a_00050'),
+           description='Performs the specific meta-analytic clustering approach included here.')
+def meta_cluster_workflow(database, output_dir=None, output_prefix=None, kernel='ALEKernel', coord=True, algorithm='kmeans', clust_range=(2,10)):
+    def VI(X, Y):
+        from math import log
+        #from https://gist.github.com/jwcarr/626cbc80e0006b526688
+        n = float(sum([len(x) for x in X]))
+        sigma = 0.0
+        for x in X:
+            p = len(x) / n
+            for y in Y:
+                q = len(y) / n
+                r = len(set(x) & set(y)) / n
+                if r > 0.0:
+                    sigma += r * (log(r / p, 2) + log(r / q, 2))
+        return abs(sigma)
+    template_file = '/Users/Katie/Dropbox/Data/scripts/pyale/pyale/resources/Grey10.nii.gz'
```

Comment:
This seems to be hardcoded to a path true only on your machine.

Reply:
Oops, I'll fix it

Reply:
Done.